### DISCUSSION TITLE
UniVRM0.81環境下でSDKのコードにエラーが出る。

### DISCUSSION THREAD

**Discussion by negipoyoc** [pixiv/vroid-sdk-developers#20] [Q&A] [answered]
# 利用環境
UniVRM 0.81が以前から安定版として広く提供されており、私が今開発しているソフトもこれを利用して現在開発を行っています。
https://github.com/vrm-c/UniVRM/wiki/ReleaseNote-v0.81.0(ja)

※今回、VRoidSDKを利用するにあたって、**Vroid SDK内部にVRM0.55が同梱されていましたがそれは私のプロジェクトにおいては不要であったので同梱されたVRMを削除したうえで利用**しようとしました。
※Unityのバージョンは2020.3.19です。

# 問題
そうした利用方法をとるにあたって、UniVRMのインポート周りの更新に追従していないためSDK内のコードにエラーが出ることがわかりました。
![image](https://user-images.githubusercontent.com/31283418/137914505-a2689595-7e6e-433f-875b-b4f480e869f2.png)

また、0.81に起因した問題ではなく、UniVRM 0.68にてランタイムインポートの仕様が変わったため、SDKがこれに追従していないことが問題だと思われます。(例えば、UniVRM0.66が入っているプロジェクトでは、エラーは発生せず正常に利用することができました。)
参考：
https://vrm-c.github.io/UniVRM/ja/vrm0/0_68_runtime_import.html#id1
https://vrm-c.github.io/UniVRM/ja/vrm0/0_79_runtime_import.html

## 質問
2点あります。

・自分が特殊な利用方法をしているかもしれませんが、VRoidSDKを利用するにあたって同梱されたVRMを使わないといけない縛りはあるでしょうか。(例えば元々のUniVRM0.55に手を加えているなどありますか？)
・今後のVRoidSDKアップデートにて、この問題は対応されますでしょうか？

何卒よろしくおねがいします。

**Comment by Haresaku** [comment, marked as answer]
> ・自分が特殊な利用方法をしているかもしれませんが、VRoidSDKを利用するにあたって同梱されたVRMを使わないといけない縛りはあるでしょうか。(例えば元々のUniVRM0.55に手を加えているなどありますか？)

現行バージョンでは特に同梱したUniVRMを利用しないといけない理由はありません。
お手数おかけしますが適宜UniVRMのアップデートをお願いいたします。

> ・今後のVRoidSDKアップデートにて、この問題は対応されますでしょうか？

こちらご報告ありがとうございます。今後のアップデートにて対応させていただきます
